_posts/2025-05-22-debezium-as-part-of-your-ai-solution.adoc

```diff
@@ -0,0 +1,438 @@
+...
+----
+
+=== Application pipeline
+
+The data pipeline is established but there is no domain specific knowledge available yet.
+We can check that model is still hallucinating.
+
+----
+$ java -jar target/quarkus-app/quarkus-run.jar query-chat Describe IterQR framework in 20 words
+
+2025-05-20 06:15:13,140 INFO  [io.deb.exa.air.RagCommand] (main) Sending query: Describe IterQR framework in 20 words
+2025-05-20 06:15:13,465 INFO  [io.deb.exa.air.MilvusRetrieavalAugmentor$RetrievalAugmentorDecorator] (main) Requested augmentation of UserMessage { name = null contents = [TextContent { text = "Describe IterQR framework in 20 words" }] }
+2025-05-20 06:15:21,521 INFO  [io.deb.exa.air.MilvusRetrieavalAugmentor$RetrievalAugmentorDecorator] (main) Result of augmentation is [DefaultContent { textSegment = TextSegment { text = "permanent" metadata = {id=permanent} }, metadata = {SCORE=0.5046951323747635, EMBEDDING_ID=permanent} }]
+2025-05-20 06:15:30,117 INFO  [io.deb.exa.air.RagCommand] (main) Chat reply: Iterative QR factorization, a permanent matrix method, breaks down data into orthogonal matrices for efficient computations.
+----
+
+The model's reply is clearly unrelated to the real framework.
+The log of `RetrievalAugmentorDecorator` shows the impact of augmentation on the message sent to the chat.
+Document `permanent` was added to the message.
+It is the dummy record from the store and as such it will be always added if it gets into top 3 matching documents.
+
+Data pipeline now processes a new domain fact (arXiv document).
+Document is downloaded from the server in textual form and inserted into PostgreSQL table.
+Debezium Server should propagate the document into Milvus store.
+
+----
+$ java -jar target/quarkus-app/quarkus-run.jar insert-document 2504.05309v1
+
+$ java -jar target/quarkus-app/quarkus-run.jar list-milvus
+
+...
+2025-05-20 06:25:00,203 INFO  [io.deb.exa.air.MilvusStore] (main) Milvus list results size: 2
+2025-05-20 06:25:00,206 INFO  [io.deb.exa.air.MilvusStore] (main) Document: {metadata={"id":"2504.05309v1"}, vector=[...], id=2504.05309v1, text=# Title
+IterQR: An Iterative Framework for LLM-based Query Rewrite in
+  e-Commercial Search System
+
+# Authors
+Shangyu Chen, Xinyu Jia, Yingfei Zhang, Shuai Zhang, Xiang Li, Wei Lin
+
+# Abstract
+The essence of modern e-Commercial search system lies in matching user's
+intent and available candidates depending on user's query, providing
+personalized and precise service. However, user's query may be incorrect due to
+ambiguous input and typo, leading to inaccurate search. These cases may be
+released by query rewr}
+2025-05-20 06:25:00,240 INFO  [io.deb.exa.air.MilvusStore] (main) Document: {metadata={"id":"permanent"}, vector=[...], id=permanent, text=permanent}...
+----
+
+Milvus store now contains the document about IterQR framework.
+Please note that by default we cut the document to 512 characters to make demo usable even on computers without GPU.
```

Comment:
```suggestion
Note that, by default, we truncate the document to 512 characters so that the demo can be used on computers that do not have a GPU.
```